src/main/java/com/google/devtools/build/lib/pkgcache/LoadingOptions.java

```diff
@@ -56,32 +58,38 @@
 
   @Option(
     name = "build_tag_filters",
-    converter = CommaSeparatedOptionListConverter.class,
+    converter = ExpressionConverter.class,
     defaultValue = "",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies a comma-separated list of tags. Each tag can be optionally "
-            + "preceded with '-' to specify excluded tags. Only those targets will be built that "
-            + "contain at least one included tag and do not contain any excluded tags. This option "
-            + "does not affect the set of tests executed with the 'test' command; those are be "
-            + "governed by the test filtering options, for example '--test_tag_filters'"
+        "Specifies a boolean expression or a list of tags separated by comma. A boolean expression must follow the "
```

Comment:
Boolean "formula", not expression.

Delete "or a list of tags separated by comma": that's the legacy syntax. Add a note at the end explaining that it is deprecated.